src/App.js

```diff
@@ -1,11 +1,124 @@
 import React, { Component } from 'react';
-import './App.css';
+import axios from 'axios';
+import MovieList from './MovieList';
+import SearchBar from './SearchBar';
 
 class App extends Component {
+  constructor() {
+    super();
+
+    this.state = {
+      movies: [],
+      savedMovies: [],
+      searchText: ''
+    };
+  }
+  componentDidMount() {
+    this.getPopularMovies();
+    this.getSavedMovies();
+  }
+  getSavedMovies() {
+    axios.get('http://localhost:4000/movies')
+      .then(resp => {
+        this.setState({
+          savedMovies: resp.data
+        });
+      });
+  }
+  getPopularMovies() {
+    const savedMoviesIdArr = this.state.savedMovies.map(savedMovie => savedMovie.id);
+    axios.get('http://api.themoviedb.org/3/movie/popular?api_key=2dba200e2682e0f8903ed87b9c9e02d1')
+      .then(resp => {
+        // Only return popular movies whos IDs are NOT found in my savedMovies
+        this.setState({
+          movies: resp.data.results.filter(resultsMovie =>
+            !savedMoviesIdArr[savedMoviesIdArr.indexOf(resultsMovie.id)])
+        });
+      });
+  }
+  handleSearchBarChange(event) {
+    this.setState({
+      searchText: event.target.value
+    });
+  }
+  handleSubmit(event) {
+    event.preventDefault();
+    const savedMoviesIdArr = this.state.savedMovies.map(savedMovie => savedMovie.id);
+    axios.get(`https://api.themoviedb.org/3/search/movie?api_key=2dba200e2682e0f8903ed87b9c9e02d1&language=en-US&query=${this.state.searchText}&page=1&include_adult=false`)
+      .then(resp => {
+        this.setState({
+          // Only return searched movies whos IDs are NOT found in my savedMovies
+          movies: resp.data.results.filter(resultsMovie =>
+            !savedMoviesIdArr[savedMoviesIdArr.indexOf(resultsMovie.id)])
+        });
+      })
+      .catch(err => {
+        console.err(`Error ${err}`);
+      });
+  }
+  handleImgError(event) {
```

Comment:
I don't think this is terrible up here, but it does feel a bit out of place. I think a better solution would be to make the movie poster its own component, and then handle this type of logic in that component.

Or if you want to pass this as a prop and make the component more generic, move this closer to, say in the `Movie` component. Up here in `App` it's just too far away and doesn't serve much of a purpose being so high up in the component tree.